Sources/Purchasing/Purchases/TransactionPoster.swift

```diff
@@ -21,16 +21,25 @@ struct PurchaseSource {
 
 }
 
+/// Encapsulates data used when posting transactions to the backend.
+struct PurchasedTransactionData {
+
+    var appUserID: String
+    var presentedOfferingID: String?
+    var unsyncedAttributes: SubscriberAttribute.Dictionary?
+    var storefront: StorefrontType?
+    var source: PurchaseSource
+
+}
+
 /// A type that can post receipts as a result of a purchased transaction.
 protocol TransactionPosterType: AnyObject, Sendable {
 
     /// Starts a `PostReceiptDataOperation` for the transaction.
     func handlePurchasedTransaction(
         _ transaction: StoreTransaction,
-        presentedOfferingID: String?,
-        storefront: StorefrontType?,
-        source: PurchaseSource,
-        completion: @escaping PurchaseCompletedBlock
+        data: PurchasedTransactionData,
+        completion: @escaping CustomerAPI.CustomerInfoResponseHandler
```

Comment:
Hmm it's a bit weird to have a reference to `CustomerAPI` in the `TransactionPoster`... I think it's ok though.

Reply:
`CustomerAPI` also has the `post(receiptData:)` method.